### ISSUE TITLE
library/core/Curl/Handler.php  Error

### ISSUE THREAD

**Issue by rsjohn** [swoole/swoole-src#3186]
1. What did you do? If possible, provide a simple script for reproducing the error.
	Set:	
```php
	Swoole\Runtime::enableCoroutine(SWOOLE_HOOK_CURL);

	Building curl header array with something like this:
	$headerArray = array(
                "POST /WebServices/abc.asmx HTTP/1.1",
                "Host: www.test.com",
                "Content-Type: text/xml; charset=utf-8",
                "Connection: close",
                "Content-Length: ". strlen($this->queryString),
                'SOAPAction: "https://www.test.com/WebServices/ "'
            );
	curl_setopt($this->_ch, CURLOPT_HTTPHEADER, $headerArray);

	The issue is this line:
		'SOAPAction: "https://www.test.com/WebServices/ "'
```
        The issue is there are two ‘:’ on that line.  The swoole cli outputs this:
PHP Notice:  Undefined offset: 1 in @swoole-src/library/core/Curl/Handler.ph  p on line 564
The raw header sent to the client at test.com is this:
```
   [ssl_verifyresult] => 0
    [scheme] =>
    [request_header] => POST /xxxxxxx HTTP/1.1
Host: xxxxxx.com:443
Content-Type: text/xml; charset=utf-8
Connection: close
SOAPAction: "https
Accept-Encoding: gzip, deflate
Content-Length: 3044


Here is the problem line: SOAPAction: "https
                                                           ^         ^
                                                      1st :   ,     2nd : and broken parse

```
2. What did you expect to see?
	
	I expected the header array to be properly parsed. Note that the normal php system curler works with the original $headerArray. The issue only happens when using: Swoole\Runtime::enableCoroutine(SWOOLE_HOOK_CURL);

3. What did you see instead?
```
	Error parsing the header: PHP Notice:  Undefined offset: 1 in @swoole-src/library/core/Curl/Handler.ph  p on line 564
```
	Incorrect soap header sent to client: SOAPAction: "https

I would love to know where this file is located? : @swoole-src/library/core/Curl/Handler.php
I suspect the fix is a simple incorrect parsing of the 2 ‘:’ but I cannot find the C++ file.  

4. What version of Swoole are you using (show your `php --ri swoole`)?
5. What is your machine environment used (including version of kernel & php & gcc) ?
```
php -r 'phpinfo();' | grep -A 20 'Swoole Team <team@swoole.com>'
Author => Swoole Team <team@swoole.com>
Version => 4.4.14
Built => Feb 19 2020 10:01:48
coroutine => enabled
epoll => enabled
eventfd => enabled
signalfd => enabled
cpu_affinity => enabled
spinlock => enabled
rwlock => enabled
sockets => enabled
openssl => OpenSSL 1.0.2k-fips  26 Jan 2017
http2 => enabled
pcre => enabled
zlib => 1.2.7
mutex_timedlock => enabled
pthread_barrier => enabled
futex => enabled
async_redis => enabled

centos-release-7-6.1810.2.el7.centos.x86_64 
Linux 4.9.27-moby #1 SMP Thu May 11 04:01:18 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux


```

**Comment by twose** [comment]
@rsjohn 首先感谢您的BUG反馈, 我是无情的自动回复机器人，请您快速确认以下几点来帮助我们定位您的问题：
1. 如你是手动升级编译的而不是首次编译, 确保你在编译前运行`make clean`清理了上次编译的内容

2. 如你使用的是Linux系统, 使用以下命令运行你的程序, 以能够复现BUG的方式操作你的程序, 并在程序运行一定时间后, 将位于/tmp目录的valgrind.log日志内容贴在此issue中

```shell

USE_ZEND_ALLOC=0 valgrind --log-file=/tmp/valgrind.log php xxx.php

```

3. 如你使用的是macOS系统，将无法使用valgrind, 请提供能够复现问题BUG的PHP代码

4. 若以上需求都无法满足，请您在此表明, 我们将快速接入人工处理

**Comment by rsjohn** [comment]
None of the suggestions from the bot apply.  This is a bad parse in the /library/core/Curl/Handler.php

**Comment by twose** [comment]
This bug has been fixed in v4.4.15, you can update to the latest version (v4.4.16)

**Comment by rsjohn** [comment]
SOLVED.
Confirmed fixed in version:
Version => 4.5.0-alpha

For those interested the code is in /php_swoole_library.h ~line 3501
